tests/container/test_heap.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import absolute_import
+import random
+import sys
+import pytest
+import six
+from tchannel.container import heap
+
+from tchannel.container.heap import HeapOperation
+
+
+class IntHeap(HeapOperation):
+    def __init__(self):
+        self.values = []
+
+    def size(self):
+        return len(self.values)
+
+    def lt(self, i, j):
+        return self.values[i] < self.values[j]
+
+    def push(self, x):
+        self.values.append(x)
+
+    def pop(self):
+        n = len(self.values)
+        item = self.values[n - 1]
+        del self.values[n - 1]
+        return item
+
+    def swap(self, i, j):
+        self.values[i], self.values[j] = self.values[j], self.values[i]
+
+
+n = 1000
+
+
+@pytest.fixture
+def int_heap():
+    return IntHeap()
+
+
+def create_values(n):
+    values = []
+    for i in six.moves.range(n):
+        values.append(i)
+
+    random.shuffle(values)
+    return values
+
+
+def verify(ph, parent):
+    child1 = 2 * parent + 1
+    child2 = 2 * parent + 2
+    if child2 < ph.size():
+        assert ph.lt(parent, child1)
+        verify(ph, child1)
+
+    if child2 < ph.size():
+        assert ph.lt(parent, child2)
+        verify(ph, child2)
+
+
+def test_init(int_heap):
+    int_heap.values = create_values(n)
+    heap.init(int_heap)
+    verify(int_heap, 0)
+
+
+def test_push(int_heap):
+    values = create_values(n)
+    for value in values:
+        heap.push(int_heap, value)
+        verify(int_heap, 0)
+
+    assert int_heap.size() == n
+
+
+def test_pop(int_heap):
+    values = create_values(n)
+    for value in values:
+        heap.push(int_heap, value)
+        verify(int_heap, 0)
+
+    for i in six.moves.range(n):
+        assert i == heap.pop(int_heap)
+
+    assert int_heap.size() == 0
+
+
+def test_remove(int_heap):
+    values = create_values(n)
+    for value in values:
+        heap.push(int_heap, value)
+        verify(int_heap, 0)
+
+    # random remove item from the heap
+    for i in six.moves.range(n - 1, -1, -1):
+        heap.remove(int_heap, random.randint(0, i))
+        verify(int_heap, 0)
+
+
+@pytest.mark.heapfuzz
+@pytest.mark.skipif(True, reason='stress test for the value heap operations')
+def test_heap_fuzz(int_heap):
```

Comment:
How long does this take to run? I'm okay with skipping it by default. We can probably convert it into a pytest-benchmark based benchmark later.